src/main/java/org/embulk/input/google_ads/GoogleAdsReporter.java

```diff
@@ -282,14 +288,92 @@ public List<String> buildWhereClauseConditions(PluginTask task, Map<String, Stri
     }
 
     public void connect()
+    {
+        this.client = buildClient(task.getLoginCustomerId().orElseGet(() -> getLoginCustomerId(task.getCustomerId())));
+    }
+
+    private Long getLoginCustomerId(String customerId)
+    {
+        try (CustomerServiceClient client = buildClient(null).getLatestVersion().createCustomerServiceClient()) {
+            return client.listAccessibleCustomers(ListAccessibleCustomersRequest.newBuilder().build())
+                    .getResourceNamesList()
+                    .stream()
+                    .map(CustomerName::parse)
+                    .map(CustomerName::getCustomerId)
+                    .map(this::getLoginCustomerClients)
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.groupingBy(LoginCustomerClient::getCustomerClientId, Collectors.maxBy(Comparator.comparing(LoginCustomerClient::getLevel))))
+                    .values()
+                    .stream()
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .filter(loginCustomerClient -> loginCustomerClient.isMe(customerId))
+                    .findFirst()
+                    .orElseThrow(() -> new RuntimeException("login customer not found [customer id: " + customerId + "]"))
+                    .getLoginCustomerId();
+        }
+    }
+
+    private List<LoginCustomerClient> getLoginCustomerClients(String customerId)
+    {
+        try (GoogleAdsServiceClient client = buildClient(Long.valueOf(customerId)).getLatestVersion().createGoogleAdsServiceClient()) {
+            return client.searchStreamCallable().call(SearchGoogleAdsStreamRequest.newBuilder()
+                            .setCustomerId(customerId)
+                            .setQuery("SELECT customer_client.id, customer_client.manager, customer_client.level FROM customer_client")
```

Comment:
Reference:
https://developers.google.com/google-ads/api/fields/v17/customer_client

```
CustomerClients only exist for manager customers.
All direct and indirect client customers are included, as well as the manager itself.
```

The `manager customers` means manager accounts (MCC).